web-platform/website/src/services/AuthenticationService.tsx

```diff
@@ -1,38 +1,73 @@
+import { onAuthStateChanged, signOut } from "firebase/auth";
 import { createContext, useContext, useEffect, useState } from "react";
 
 import { User } from "../models/User";
+import { useFirebase } from "./FirebaseService";
+import { signupWithEmail as signupWithEmailFunction } from "./UserAccountService";
 
-export type UserDataContext = {
+export interface UserDataContext {
   isLoaded: boolean;
   user?: User;
   isLoggingIn: boolean;
+  isUserVerified: boolean;
   logout: () => Promise<void>;
-};
+  signupWithEmail(email: string, password: string): Promise<void>;
+}
 
 const AuthenticationContext = createContext<UserDataContext>({
-  isLoaded: true,
+  isLoaded: false,
   user: undefined,
   isLoggingIn: false,
-  logout: async () => {},
-});
+  isUserVerified: false,
+} as UserDataContext);
 
 export function useAuthentication() {
   return useContext(AuthenticationContext);
 }
 
 export function AuthenticationProvider(props: { children: React.ReactNode }) {
-  const [user, setUser] = useState<User>();
+  const [user, setUser] = useState<User | undefined>();
+  const [isUserVerified, setIsUserVerified] = useState(false);
   const [isLoggingIn] = useState(false);
   const [isLoaded, setIsLoaded] = useState(false);
 
+  const { auth } = useFirebase();
+
   useEffect(() => {
-    // TODO: Retrieve user here...
+    onAuthStateChanged(auth, (firebaseUser) => {
```

Comment:
I think you should add `return` in front of this function call. `onAuthStateChanged` returns its own unsubscribe function, and returning that tells the React Effect to call the unsubscribe function when the provider is being unloaded

Reply:
Agreed (even though the auth context lifecycle is at the app level). Doing it next PR.